### PR TITLE
rename typespec and clear generated samples

### PR DESCRIPTION
**Pull request by chunyu3**
# Description

- rename cadl to typespec
- clear autogen samples when `clear-output-folder` is true

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

**Comment by pshao25** [comment]
I would suggest change `cadl.json` to `typespec.json`. `tsp` is another file extension name. It is uncomfortable to me to change to `tsp.json`.

**Comment by chunyu3** [comment]
> typespec

Maybe we change it to tspCodeModel.json to indicate that it is a intermediate code model from typespec, seminar to CodeModel.yaml from swagger.

File: src/AutoRest.CSharp/LowLevel/AutoRest/DpgOutputLibrary.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@ internal class DpgOutputLibrary : OutputLibrary
     {
         private readonly IReadOnlyDictionary<InputEnumType, EnumType> _enums;
         private readonly IReadOnlyDictionary<InputModelType, ModelTypeProvider> _models;
-        private readonly bool _isCadlInput;
+        private readonly bool _isTypeSpecInput;
```
Comment (lirenhe) [inline src/AutoRest.CSharp/LowLevel/AutoRest/DpgOutputLibrary.cs:19]: I found in some place,  tsp is used in variable names (tspOutputFileName) and in other places, typespec are used like _isTypeSpecInput.
Can we make it consistent in .ts code?

**Review by ArcturusZhang — COMMENTED** [review]
Could we align the words in our source code? There are existing code using `Typespec`, and some of them are using `TypeSpec`. Maybe we could unify them by changing both of them to `tsp`?
I could go either way but we are better to unify than diverge

**Comment by chunyu3** [comment]
> Typespec

When use lowcase it should be `typespec`, if capitalize it , it should be `TypeSpec`, this is the rule from typesepc team.

**Comment by chunyu3** [comment]
@lirenhe @ArcturusZhang  I update the `typespec` word in the source code to comply the rule mentioned above. Please review again. Thanks

**Review by ArcturusZhang — APPROVED** [review]

**Comment by lirenhe** [comment]
@chunyu3,  could you resolve the conflict in this PR first?

File: src/AutoRest.CSharp/LowLevel/AutoRest/DpgOutputLibrary.cs [hunk 2]
```diff
@@ -19,7 +19,7 @@ internal class DpgOutputLibrary : OutputLibrary
         private readonly string _rootNamespace;
         private readonly IReadOnlyDictionary<InputEnumType, EnumType> _enums;
         private readonly IReadOnlyDictionary<InputModelType, ModelTypeProvider> _models;
-        private readonly bool _isCadlInput;
+        private readonly bool _isTspInput;
```
Comment (lirenhe) [inline src/AutoRest.CSharp/LowLevel/AutoRest/DpgOutputLibrary.cs:22]: Can we use  _isTypeSpecInput here?

Comment (chunyu3) [reply src/AutoRest.CSharp/LowLevel/AutoRest/DpgOutputLibrary.cs:22]: We will use tspOutput, tspConfig in source code, for consistent, I prefer to using isTspInput also.

**Review by lirenhe — APPROVED** [review]

**Pull request merged**